docs/reference/latency.md

```diff
@@ -0,0 +1,96 @@
+
+## Local execution
+
+In general, calls to the Chainweb `/local` endpoint return results almost immediately.
+Depending on how long it takes to execute the Pact code in a transaction, you can typically expect to see results from `/local` call within seconds, at most. 
+However, it's important to know that `/local` calls are executed on a single chain and served in a single-threaded, first-come-first-served basis.
+If a node is under contention—because it's catching up with state, busy processing multiple `/local` calls, or performing mempool checks—execution could take as long as needed for the request to be serviced in the queue.
+
+## Block production
+
+On average, producing a block takes 30 seconds.
+However, the average block production time represents the exponential distribution between events.
+The actual time it takes to produce any given block depends on the time required to prepare the block for mining.
+Before a block can be mined on any chain, it must satisfy specific requirements.
+
+To give you better insight into how blocks are produced, the following example describes the workflow and requirements for mining a block.
+For this example, the target chain for the block is chain 1 on a Kadena network with 20 chains.
+
+To start block production on chain 1, the new block candidate must provide the following inputs:
+
+- The block header hash for a block that exists on chain 1 to serve as the parent of the new block.
+- The block headers for the parent block of the new block on the adjacent chains, which are currently chains 6, 11, and 18 for chain 1.
+- The sequence of transactions in the new block and the output from running the transactions in the new block.
+- The nonce and creation time for the new block, which are set by the miner.
+
+### Block parents and orphans
+
+If the block selected to serve as the parent of the new candidate block is superseded by the arrival of a new block on the target chain, the block production process must start over with the new parent block.
+If the block production process starts over, the new block candidate that was in processed is no longer valid for the continuity of the chian and is deemed an **orphaned** block.
+
+### Block height and payload
+
+The next step in the block production process involves two requirements that are satisfied in parallel. 
+
+- The first requirement is to determine whether the chains that are adjacent to the target chain in the chain graph have any blocks at the same height as the parent block.
+  If the adjacent chains have blocks at the same height as the parent block, those chains are deemed ready to add a new block and satisfy the braiding requirement of the Chainweb protocol.
+
+- The second requirement is to produce the block payload on the target chain.
+  Producing the payload involves querying the mempool for a set of transactions that can fit in the block, and executing those transactions on the Pact state.
```

Comment:
```suggestion
- The second requirement is to produce the block payload on the target chain.
  Producing the payload involves querying the mempool for a set of transactions that can fit in the block, and executing those transactions on the Pact state. Currently, the gas limit for a block is 150000, and each unit gas may represent up to 2.5 microseconds of execution time, leading to a maximum execution time of 375 milliseconds, but in practice due to node contention this may be higher, up to 600 milliseconds.
```

Reply:
Making sure that the time maximum here is explicit and explained.